### PR TITLE
feat: enhance ButtonInteractionHandler with comprehensive testing and click behavior fixes

### PR DESCRIPTION
**Pull request by MasatoMakino**
## Summary
- Comprehensive test suite expansion for ButtonInteractionHandler
- Fix click behavior for down→out→up sequences to match standard UI patterns  
- Enhanced test robustness with event order verification and payload validation

## Test Coverage Improvements
- **From 3 to 27 tests** covering all interaction scenarios
- **96.72% statement coverage** (improved from 88.52%)
- **100% function coverage** achieved
- Systematic 6-phase test improvement methodology applied

## Click Behavior Fix
**Problem**: ButtonInteractionHandler emitted clicks for down→out→up sequences, contradicting standard UI expectations

**Solution**: Reset `_isPress` state on mouse out events to prevent unintended clicks when releasing outside object boundaries

**Impact**: More intuitive interaction behavior aligned with DOM/React standards

## Enhanced Test Quality
- **False positive verification** for all new tests
- **Event emission order validation** using `invocationCallOrder`
- **Comprehensive payload shape checks** with `objectContaining`
- **Integration testing** for complex interaction sequences
- **Type-safe assertions** eliminating all `as any` casts

## Test Categories Added
1. **Basic Interaction** - Core mouse event handling
2. **State Management** - Enable/disable and frozen states  
3. **Boundary Conditions** - Edge cases and error handling
4. **Integration Scenarios** - Complex interaction sequences

## Robustness Improvements
- Watch-mode development with 60x performance improvement
- Systematic false positive testing for reliability
- Eliminated test responsibility overlaps with other components
- Added explanatory comments for unreachable code paths

## Files Modified
- `src/interactionHandler/ButtonInteractionHandler.ts` - Click behavior fix and documentation
- `__test__/ButtonInteractionHandler.spec.ts` - Comprehensive new test suite
- `__test__/ButtonInteractionHandler.eventObject.spec.ts` - Enhanced event validation

🤖 Generated with [Claude Code](https://claude.ai/code)

## Summary by CodeRabbit

* **New Features**
  * Added comprehensive tests validating interactive 3D mesh interactions, state transitions, and integration scenarios.

* **Bug Fixes**
  * Prevented unintended clicks by clearing press/hover state when pointer leaves, when disabling, or when an object is frozen.

* **Tests**
  * Improved test clarity and rigor with explicit descriptions, clearer assertions, event-order checks, and verification that event payloads preserve custom values.

### CODE REVIEW

File: __test__/ButtonInteractionHandler.eventObject.spec.ts [hunk 1]
```diff
@@ -46,18 +61,38 @@ describe("ButtonInteractionHandler Event Object Validation", () => {
...
 
-    expect(upEvent).not.toBe(clickEvent);
-    expect(upEvent.type).toBe("up");
-    expect(clickEvent.type).toBe("click");
+    // Ensure 'up' event is emitted before 'click'
+    // Note: verify your Vitest version supports `mock.invocationCallOrder`.
+    expect(upSpy.mock.invocationCallOrder[0]).toBeLessThan(
+      clickSpy.mock.invocationCallOrder[0],
+    );
+
```
[the remark this answers is not in the document]
Comment (MasatoMakino) [reply __test__/ButtonInteractionHandler.eventObject.spec.ts:81]: https://vitest.dev/api/mock.html#mock-invocationcallorder
v3.2.4で対応しています。
